spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestAggregatePushDown.java

```diff
@@ -249,6 +250,78 @@ public void testAggregateNotPushDownIfOneCantPushDown() {
     assertEquals("expected and actual should equal", expected, actual);
   }
 
+  @Test
+  public void testAggregationPushdownStructInteger() {
+    testAggregationPushdownStruct(
+        2L,
+        3L,
+        2L,
+        "(id BIGINT, struct_with_int STRUCT<c1:BIGINT>)",
+        "struct_with_int.c1",
+        "(1, named_struct(\"c1\", NULL))",
+        "(2, named_struct(\"c1\", 2))",
+        "(3, named_struct(\"c1\", 3))");
+  }
+
+  @Test
+  public void testAggregationPushdownNestedStruct() {
+    testAggregationPushdownStruct(
+        2L,
+        3L,
+        2L,
+        "(id BIGINT, struct_with_int STRUCT<c1:STRUCT<c2:STRUCT<c3:STRUCT<c4:BIGINT>>>>)",
+        "struct_with_int.c1.c2.c3.c4",
+        "(1, named_struct(\"c1\", named_struct(\"c2\", named_struct(\"c3\", named_struct(\"c4\", NULL)))))",
+        "(2, named_struct(\"c1\", named_struct(\"c2\", named_struct(\"c3\", named_struct(\"c4\", 2)))))",
+        "(3, named_struct(\"c1\", named_struct(\"c2\", named_struct(\"c3\", named_struct(\"c4\", 3)))))");
+  }
+
+  @Test
+  public void testAggregationPushdownStructTimestamp() {
+    long timestamp = System.currentTimeMillis();
+    long futureTimestamp = timestamp + 5000;
+    Timestamp expectedMax = new Timestamp(futureTimestamp / 1000 * 1000);
+    Timestamp expectedMin = new Timestamp(1000 * (timestamp / 1000));
```

Comment:
Why are the expressions different other than the timestamp variable?

Reply:
I was aiming to test multiple data types. The issue with converting in Spark, is that the goal of the test is to verify that there was indeed aggregation pushdown and using any conversion functions will eliminate the pushdown that's performed . 

In the latest version in Spark I insert records explicitly in UTC time (which will then be stored in timestamp without timezone format in Iceberg). Then when the records are read, the `java.sql.Timestamp` will be a point in time, milliseconds since epoch. This would be deterministic across time zones unless there's some implicit conversion that I'm missing.